### PR TITLE
🏃Return error if instance go error

### LINKED ISSUE #968: Instance with error state cann't be deleted timely
/kind bug

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]
1. Create a workload cluster.
2. Master instance go error due to some exception in nova.
3. Delete this cluster.
4. Instance could not be deleted in 30 minites.

**What did you expect to happen:**
Instance could be deleted timely.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]
I debug code and found that reconciler were waiting for instance to be active until timeout(default 30 minitues), and this blocks the reconciler, so deletion would not be triggered until creation timeout.
My suggestion is: If state of instance is error, return and stop waiting.

**Environment:**

- Cluster API Provider OpenStack version (Or `git rev-parse HEAD` if manually built): 0.4.0
- Cluster-API version: 0.4.0
- OpenStack version: Rocky
- Minikube/KIND version:
- Kubernetes version (use `kubectl version`): 1.20.10
- OS (e.g. from `/etc/os-release`): centos-8

### PR DESCRIPTION
**Pull request by zhangdb-git**
**What this PR does / why we need it**:
If instance go error due to some exceptions of openstack, we can return error at once and not nesscessary to wait for timeout.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #968

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**TODOs**:

- [ ] squashed commits
- if necessary:
  - [ ] includes documentation
  - [ ] adds unit tests

/hold

### CODE REVIEW

**Comment by tobiasgiese** [comment]
Thanks for contributing @zhangdb-git
Could you please create an issue to this PR? :)
Thanks

/ok-to-test

File: pkg/cloud/services/compute/instance.go [hunk 1]
```diff
@@ -276,6 +276,9 @@ func (s *Service) createInstance(eventObject runtime.Object, clusterName string,
 			}
 			return false, err
 		}
+		if createdInstance.State == infrav1.InstanceStateError {
+			return false, fmt.Errorf("openstack instance %s go error.", createdInstance.ID)
```
Comment (tobiasgiese) [inline pkg/cloud/services/compute/instance.go:280]:
```suggestion
			return false, fmt.Errorf("error creating OpenStack instance %s, status changed to error", createdInstance.ID)
```

Maybe a bit more convincing error message

**Comment by zhangdb-git** [comment]
I reported a bug:
https://github.com/kubernetes-sigs/cluster-api-provider-openstack/issues/968

**Review by chrischdi — COMMENTED** [review]
Although I don't like the `fmt.Errorf("error ` pattern (an `error` type is always an error ;-) ) we already do it the same above.

/lgtm

**Pull request merged**